### ISSUE TITLE
VStack and Button not working properly

### ISSUE THREAD

**Issue by gerzonc** [andrew-levy/swiftui-react-native#36]
Hi! I've got to refactor some of my components since I was using v5, and I noticed 2 issues:

### Button not respecting modifier order

<img src="https://github.com/andrew-levy/swiftui-react-native/assets/36211892/8c553608-b55c-40d3-a070-776ac5cddb79"  height="450px" />

## Expected behavior
Should be similar to SwiftUI for the VStack 
```swift
VStack()
  .padding()
  .frame(maxWidth: .infinity, maxHeight: .infinity)
  .background(.black)
```

For the Button, it should respect the order as well, but it doesn't (because of this I can't have full width buttons, it seems)

My code:

```js
 const { height, width } = useWindowDimensions();
  const { stateNavigator } = useContext(NavigationContext);

  return (
    <VStack padding frame={{ width, height }} background="black">
      <Image systemName="face.smiling" fontSize={50} />
      <Text font="title" foregroundStyle="white">
        Hello World!
      </Text>
      <Button
        title="Profile"
        tint="purple"
        buttonStyle="borderedProminent"
        frame={{ width }}
        action={() => stateNavigator.navigate('profile')}
      />
    </VStack>
  );
```

**Comment by andrew-levy** [comment]
@gerzonc thank you for testing! I'll look into these asap

**Comment by andrew-levy** [comment]
I'm able to reproduce both of these. I'm struggling to find the cause of them but will need some more time to dig in. Thanks again for pointing it out.

**Comment by gerzonc** [comment]
> I'm able to reproduce both of these. I'm struggling to find the cause of them but will need some more time to dig in. Thanks again for pointing it out.

Sure thing! No problem! I'll still continue building the entire app in the current state to check if I can find any other issues!

**Comment by andrew-levy** [comment]
I figured out what's causing vstack issue. I have a workaround for that let me see what I can do. Still looking into the button issue

**Comment by andrew-levy** [comment]
As for the button issue, it turns out that's not how things work in SwiftUI (see [here](https://stackoverflow.com/questions/56471004/making-button-span-across-vstack)). If you wanted the button to take a certain width, you need to apply the frame to the button content. Unfortunately thats not supported currently, you can only pass a string to the `title` prop. Might add this soon tho, if i can get it working.

**Comment by andrew-levy** [comment]
@gerzonc Both of these should be fixed in swiftui-react-native@6.1.0. Let me know if you still run into issues.

1. For the VStack issue, you will need to do this in a more ~swiftui way~. See [here](https://stackoverflow.com/questions/56487323/make-a-vstack-fill-the-width-of-the-screen-in-swiftui) for an example. Essentially you'll want to use something like:
```tsx
<VStack 
  frame={{ maxWidth: "infinity", maxHeight: "infinity" }} 
  ignoresSafeArea={true}
  background="black"
 />
```

NOTE: the order of these props matters a lot! Applying the frame before the background color is different from the background color first then the frame.

2. As for the Button issue, I added initial support for passing children to a button. If you want a full width button, you need to apply maxWidth: "infinity" to the children of the button as shown [here](https://sarunw.com/posts/swiftui-button-style-width/). This is the only way to do it in swiftui.
```tsx
<Button
  tint="purple"
  buttonStyle="borderedProminent"
  action={() => stateNavigator.navigate('profile')}
>
  <Text frame={{ maxWidth: "infinity"}}>Profile</Text>
</Button> 
```
